### PR TITLE
[rollout, ckpt] fix: scope SGLang delta masked copies

### PR DESCRIPTION
**Pull request by zhangxin81**
### What does this PR do?

The `delta_sharded` SGLang loader applies sparse updates by temporarily patching `Tensor.copy_`: NaN positions mean "keep the current model value."

The patch is process-global while `model.load_weights()` runs. Previously, any same-shaped floating-point copy in that call inherited the masked semantics, including writes into scratch tensors or derived post-load state.

This PR scopes masked overwrite to storage owned by the model's registered parameters and buffers. Parameter/buffer views remain covered because the check uses the underlying storage identity. Copies into temporary tensors use ordinary `copy_`.

It also isolates post-load hooks that `model.load_weights()` may invoke:

- root `model.post_load_weights`;
- unique `quant_method.process_weights_after_loading` hooks.

Those hooks run with the original `Tensor.copy_`, then the sparse loader patch is reinstated until the surrounding weight load finishes. All patches and hooks are restored after success or exceptions.

This implements the masked-copy safety guard documented as a known issue in #7060, which is part of the rollout weight-sync roadmap linked by #6985.

### Checklist Before Starting

- [x] Read roadmap #6985 and tracking issue #7060.
- [x] Inspected all open PRs and searched for `masked copy delta loader`, `process_weights_after_loading delta`, and `7060 in:body`.
- [x] No open PR implements parameter-storage scoping or post-load hook isolation.
- [x] #7085 is adjacent but not a duplicate. It adds block placement, VeOmni FSDP2+EP export, and a sync-free `torch.where` masked-copy kernel. This PR changes the safety boundary around that kernel. A compatibility worktree retained #7085's kernel, applied these guards, and passed all 12 focused tests.
- [x] The title follows the repository format: `[rollout, ckpt] fix: scope SGLang delta masked copies`.

### Test

Focused and related CPU checkpoint-engine tests:

```bash
PYTHONPATH=$PWD /tmp/verl-delta-loader-venv/bin/python -m pytest -q \
  tests/checkpoint_engine/test_sglang_loader.py \
  tests/checkpoint_engine/test_sharded_delta.py \
  tests/checkpoint_engine/test_global_steps_on_cpu.py \
  tests/checkpoint_engine/test_fsdp_lora_only_checkpoint_on_cpu.py
```

Result:

```text
35 passed
```

The focused SGLang loader file contains 12 tests covering:

- sparse delta bit identity and untouched positions;
- model parameter views and registered buffers;
- scratch tensors retaining ordinary NaN-copy behavior;
- root-model and quantization post-load hooks using ordinary `copy_`;
- restoration after apply errors;
- rollback when a hook cannot be safely wrapped;
- fail-closed behavior when model storage cannot be identified;
- checksum mismatch and dense-seed behavior.

Full repository checks:

```bash
pre-commit run --all-files --show-diff-on-failure
```

Result: all hooks passed.

The complete checkpoint-engine directory additionally includes GPU/NPU/model and live-server tests. On this CPU/macOS host, its three hardware/model-path failures and one missing `ROLLOUT_NAME` setup error were reproduced unchanged on `origin/main@2423f801`.

### API and Usage Example

No public API, configuration, or wire-format change. Existing `delta_sharded` + SGLang configurations continue to register:

```text
verl.workers.rollout.sglang_rollout.delta_loader.apply_delta
```

This PR does not claim general quantized delta-rollout support. The custom loader API receives one flush at a time without the final-flush bit, so quantization hooks that must run once after a complete sync still require the separate quantized-rollout protocol work tracked in #7060.

### Design & Code Changes

- Build one set of `(device, storage pointer, storage bytes)` identities from registered model parameters and buffers per delta flush.
- Apply NaN-masked overwrite only when the destination shares one of those storages.
- Preserve masked semantics for narrowed TP parameter/buffer views.
- Keep temporary and scratch copies on the original `Tensor.copy_` path.
- Temporarily restore `Tensor.copy_` while model-invoked post-load hooks run.
- Restore all global and instance hooks in `finally` blocks.
- Fail closed if model storage or a post-load hook cannot be safely isolated.

A synthetic CPU overhead check with 2,000 model parameters measured:

```text
guarded parameter copy: 7.128 us/op
guarded scratch copy:   1.327 us/op
storage index build:    1.231 ms/flush
```

This is overhead evidence only, not a training-throughput claim.

### Checklist Before Submitting

- [x] Read `CONTRIBUTING.md` and repository `AGENTS.md`.
- [x] Duplicate-work checks completed.
- [x] Focused and broader CPU tests passed.
- [x] Full pre-commit passed.
- [x] No generated config, recipe, or documentation surface is affected.
- [x] The submitter reviewed every changed line and personally confirmed the test and compatibility evidence before submission.

### AI assistance

OpenAI Codex assisted with roadmap analysis, implementation, tests, and PR drafting. The human submitter reviewed every changed line, understands the process-global monkeypatch and post-load boundaries, and personally confirmed the validation evidence before submission.

### CODE REVIEW

**Comment by wuxibin89** [comment]
@ChangyiYang Please take a look.

**Comment by ChangyiYang** [comment]
@zhangxin81 hi thanks for taking a look on this. Have you run any e2e test?
